api/types/resource.go

```diff
@@ -201,6 +227,11 @@ func (m *Metadata) CheckAndSetDefaults() error {
 		}
 	}
 
+	// Check the origin value.
+	if !utils.SliceContainsStr(append(OriginValues, ""), m.Origin()) {
+		return trace.BadParameter("invalid origin value: %q", m.Origin())
```

Comment:
Nit: We usually try to give possible accepted values in the error message in such cases, if that'd make sense here.

Reply:
Done, added a note to the message that this occurring amounts to a bug.